### PR TITLE
Bld 28198 - Async with restore and missed messages fetching support

### PR DESCRIPTION
**Pull request by tehilaR2**

### CODE REVIEW

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+    /**
+     * Save provided chat details to the shared preferences.
+     */
+    fun saveAccount(restore: Boolean, apiKey: String, appId: String, senderId: Long? = null, lastMessage: String? = null) {
+        sharedPrefs.run {
+
+            val canRestore = restore && this@AsyncAccountRecovery.apiKey.equals(apiKey)
+
+            edit().apply {
```
Comment (ozbenoved) [inline SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt]: "edit" is unclear - what is it? maybe you should change its name

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/BoldChat.kt [hunk 2]
```diff
@@ -23,6 +23,6 @@ open class BoldChat : BasicChat() {
         val formalBoldAccount = BoldAccount("2300000001700000000:2278936004449775473:sHkdAhpSpMO/cnqzemsYUuf2iFOyPUYV") // Mobile
         val fameBoldAccount = BoldAccount("2300000001700000000:2279148490312878292:grCCPGyzmyITEocnaE+owvjtbasV16eV") // Fame
 
-        val defaultBoldAccount = formalBoldAccount
+        val defaultBoldAccount = fameBoldAccount
```
Comment (ozbenoved) [inline SDKSamples/app/src/main/java/com/sdk/samples/topics/BoldChat.kt]: why do we have "fame and formal" bold accounts here? 
I thought we deleted them...

it should be only the default demo account.

Comment (tehilaR2) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/BoldChat.kt]: it's the develop branch.
but it should point to the formal one

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt (hunk 1 above)
Comment (tehilaR2) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt]: by edit() u open the shared preferences to write and the apply writes things on it.
why isn't it clear?

Comment (ozbenoved) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt]: ok

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/BoldChat.kt (hunk 2 above)
Comment (ozbenoved) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/BoldChat.kt]: ok

**Pull request merged**